### PR TITLE
Mvp implementation

### PR DESCRIPTION
**Pull request by mariano98montero**
MVP implementation
Added contract, model, presenter and view folders. 
ParkingModel, MainActivityContract,  ParkingPresenter, ActivityView and ParkingView classes and interfaces.
Added new String in string resources for the Toast message.
mvp_implementation_base_structure
mvp_implementation_device
Device: Pixel_3a.

### CODE REVIEW

File: app/src/main/java/com/example/parkingsystemjava/mvp/model/ParkingModel.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    @Override
+    public void setParkingLots() {
+        this.parkingLots = 6;
+    }
+
+    @Override
+    public int getParkingLots() {
+        return parkingLots;
+    }
+
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/parkingsystemjava/mvp/model/ParkingModel.java]: remove unnecessary empty space

File: app/src/main/java/com/example/parkingsystemjava/mvp/contract/MainActivityContract.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.example.parkingsystemjava.mvp.contract;
+
+public interface MainActivityContract {
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/parkingsystemjava/mvp/contract/MainActivityContract.java]: pls rename the name of the contrat to ParkingContract

File: app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingPresenter.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import com.example.parkingsystemjava.mvp.contract.MainActivityContract;
+import com.example.parkingsystemjava.mvp.model.ParkingModel;
+import com.example.parkingsystemjava.mvp.view.ParkingView;
+
+public class ParkingPresenter implements MainActivityContract.MainActivityPresenter {
+
+    private MainActivityContract.MainActivityModel model;
+    private MainActivityContract.MainActivityView view;
+
+    public ParkingPresenter(ParkingModel model, ParkingView view) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingPresenter.java]: here you should receive the contracts as parameters

File: app/src/main/java/com/example/parkingsystemjava/mvp/view/ActivityView.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package com.example.parkingsystemjava.mvp.view;
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystemjava/mvp/view/ActivityView.java]: this class should be into a package "base" inside the view package

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/parkingsystemjava/mvp/model/ParkingModel.java (hunk 1 above)
Comment (mariano98montero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/model/ParkingModel.java]: done!

File: app/src/main/java/com/example/parkingsystemjava/mvp/contract/MainActivityContract.java (hunk 2 above)
Comment (mariano98montero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/contract/MainActivityContract.java]: done!

File: app/src/main/java/com/example/parkingsystemjava/mvp/view/ActivityView.java (hunk 4 above)
Comment (mariano98montero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/view/ActivityView.java]: done!

File: app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingPresenter.java (hunk 3 above)
Comment (mariano98montero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingPresenter.java]: done!

**Review by mauroLonghi — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/example/parkingsystemjava/mvp/view/base/ActivityView.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
+package com.example.parkingsystemjava.mvp.view.base;
+
+import android.app.Activity;
+import android.app.FragmentManager;
+import android.content.Context;
+
+import androidx.annotation.Nullable;
+
+import java.lang.ref.WeakReference;
```
Comment (nicolbidone) [inline app/src/main/java/com/example/parkingsystemjava/mvp/view/base/ActivityView.java:7]: please remember not to use line breaks between imports

File: app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingView.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
+package com.example.parkingsystemjava.mvp.view;
+
+import android.app.Activity;
+import android.content.Context;
+import android.widget.Toast;
+
+import com.example.parkingsystemjava.R;
```
Comment (nicolbidone) [inline app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingView.java:6]: please remember not to use line breaks between imports

**Review by nicolbidone — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Pull request closed without merging**